### ISSUE TITLE
Bug: Compress image manipulation removes png image transparency

### ISSUE THREAD

**Issue by ivanz040592** [codeigniter4/CodeIgniter4#4178]
I am using compress image from codeigniter 4 library with below code :

```php
\Config\Services::image()->withFile($this->request->getFile('storeimage'))->reorient()->save('images/store/test.png', 70)
```

But it removes transparency png image. Can i compress png image without losses the transparency in ci 4?

I put below code in GDHanlder.php (save function) but it makes compress image not working, image size result still same and not compressed

```php
imageAlphaBlending($this->resource, true);
imageSaveAlpha($this->resource, true);
```

```php
case IMAGETYPE_PNG:
    imageAlphaBlending($this->resource, true);
    imageSaveAlpha($this->resource, true);

    if (! function_exists('imagepng'))
    {
        throw ImageException::forInvalidImageCreate(lang('images.pngNotSupported'));
    }

    if (! @imagepng($this->resource, $target))
    {
        throw ImageException::forSaveFailed();
    }
    break;
```

**Comment by kenjis** [comment]
@ivanz040592 
Sorry for the delay in responding.

Try this:
```diff
--- a/system/Images/Handlers/GDHandler.php
+++ b/system/Images/Handlers/GDHandler.php
@@ -251,6 +251,9 @@ class GDHandler extends BaseHandler
                     throw ImageException::forInvalidImageCreate(lang('Images.pngNotSupported'));
                 }
 
+                imagealphablending($this->resource, false);
+                imagesavealpha($this->resource, true);
+
                 if (! @imagepng($this->resource, $target)) {
                     throw ImageException::forSaveFailed();
                 }
```

**Comment by kenjis** [comment]
This bug was fixed in the `develop` branch.
And it would be included in v4.2.0.
